internal/cli/atlas/security/ldap/save.go

```diff
@@ -0,0 +1,109 @@
+	bindUsername          string
+	bindPassword          string
+	caCertificate         string
+	authzQueryTemplate    string
+	authenticationEnabled bool
+	authorizationEnabled  bool
+	store                 store.LDAPConfigurationSaver
+}
+
+func (opts *SaveOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+var saveTemplate = `HOSTNAME	PORT	AUTHENTICATION	AUTHORIZATION
+{{.LDAP.Hostname}}	{{.LDAP.Port}}	{{.LDAP.AuthenticationEnabled}}	{{.LDAP.AuthorizationEnabled}}
+`
+
+func (opts *SaveOpts) Run() error {
+	r, err := opts.store.SaveLDAPConfiguration(opts.ConfigProjectID(), opts.newLDAPConfiguration())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *SaveOpts) newLDAPConfiguration() *atlas.LDAPConfiguration {
+	return &atlas.LDAPConfiguration{
+		LDAP: &atlas.LDAP{
+			AuthenticationEnabled: opts.authenticationEnabled,
+			AuthorizationEnabled:  opts.authorizationEnabled,
+			Hostname:              opts.hostname,
+			Port:                  opts.port,
+			BindUsername:          opts.bindUsername,
+			BindPassword:          opts.bindPassword,
+			CaCertificate:         opts.caCertificate,
+			AuthzQueryTemplate:    opts.authzQueryTemplate,
+		},
+	}
+}
+
+// mongocli atlas security ldap save --hostname hostname --port port --bindUsername bindUsername --bindPassword bindPassword --caCertificate caCertificate
+// --authzQueryTemplate authzQueryTemplate --authenticationEnabled authenticationEnabled --authorizationEnabled authorizationEnabled [--projectId projectId]
+func SaveBuilder() *cobra.Command {
+	opts := &SaveOpts{}
+	cmd := &cobra.Command{
+		Use:   "save",
+		Short: save,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(
+				opts.initStore,
+				opts.InitOutput(cmd.OutOrStdout(), saveTemplate))
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.hostname, flag.Hostname, "", usage.LDAPHostname)
+	cmd.Flags().IntVar(&opts.port, flag.Port, 636, usage.LDAPPort)
+	cmd.Flags().StringVar(&opts.bindUsername, flag.BindUsername, "", usage.BindUsername)
+	cmd.Flags().StringVar(&opts.bindPassword, flag.BindPassword, "", usage.BindPassword)
+	cmd.Flags().StringVar(&opts.caCertificate, flag.CaCertificate, "", usage.CaCertificate)
+	cmd.Flags().StringVar(&opts.authzQueryTemplate, flag.AuthzQueryTemplate, "", usage.AuthzQueryTemplate)
+	cmd.Flags().BoolVar(&opts.authenticationEnabled, flag.AuthenticationEnabled, false, usage.AuthenticationEnabled)
+	cmd.Flags().BoolVar(&opts.authorizationEnabled, flag.AuthorizationEnabled, false, usage.AuthorizationEnabled)
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
+
+	_ = cmd.MarkFlagRequired(flag.Hostname)
+	_ = cmd.MarkFlagRequired(flag.BindUsername)
+	_ = cmd.MarkFlagRequired(flag.BindPassword)
```

Comment:
[q] The port is a required flag on the ticket - I see a default value is specified above. I'm assuming as a result, we don't need to mark as required here.

Reply:
yes, when we set the default value we don't need to make the flag required